lib/src/requests.dart

```diff
@@ -83,14 +83,9 @@ class Requests {
     for (var key in responseHeaders.keys) {
       if (Common.equalsIgnoreCase(key, 'set-cookie')) {
         String cookie = responseHeaders[key];
-        cookie.split(",").forEach((String one) {
-          one
-              .split(";")
-              .map((x) => x.trim().split("="))
-              .where((x) => x.length == 2)
-              .where((x) => !_cookiesKeysToIgnore.contains(x[0].toLowerCase()))
-              .forEach((x) => cookies[x[0]] = x[1]);
-        });
+        RegExp regExp = RegExp(r"([A-Za-z0-9_]*)=([A-Za-z0-9_=.-]*)");
+        var matches = regExp.allMatches(cookie).toList();
+        cookies[matches[0].group(1)] = matches[0].group(2);
```

Comment:
please add a check to cover cases where the value does not respect the rules
imagine this is the cookie's value
```
123
```
`matches` will be empty in this case

Reply:
@jossef added with latest commit. A check to see if matches.isNotEmpty before indexing into the first item.